src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -738,6 +957,24 @@ private BoundExpression CreateAndPopulateList(BoundCollectionExpression node, Ty
                         // list.Add(element);
                         expressions.Add(
                             _factory.Call(listTemp, addMethod, rewrittenValue));
+                    },
+                    (sideEffects, spanTemp, spreadElement, rewrittenSpreadOperand) =>
+                    {
+                        if (addRangeMethod is null)
+                            return false;
+
+                        var type = rewrittenSpreadOperand.Type!;
+
+                        var useSiteInfo = GetNewCompoundUseSiteInfo();
+                        var conversion = _compilation.Conversions.ClassifyConversionFromType(type, addRangeMethod.Parameters[0].Type, isChecked: false, ref useSiteInfo);
+                        _diagnostics.Add(rewrittenSpreadOperand.Syntax, useSiteInfo);
+                        if (conversion.IsImplicit && conversion.IsReference)
```

Comment:
>conversion.IsReference [](http://example.com/codeflow?start=53&length=22)

Are we testing cases other than reference conversions? For instance:
- Identity conversion, where the spread collection is an `IEnumerable<T>`
- Variance conversion, where the spread collection is an `IEnumerable<U> where U : T`
- Boxing conversion, where the spread collection is a `struct`